### PR TITLE
Find non-duplicates in an array

### PR DESCRIPTION
**Pull request by jsca-kwok**
This PR proposes to add solutions for finding non-duplicate elements in an array:
- using a hash map
- using .count()

### CODE REVIEW

File: data_structures/arrays/single_number.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+puts(single_number(nums))
+# Output: 4
+nums = [1]
+puts(single_number(nums))
+# Output: 1
+
+#
+# Approach 2: Use Ruby .count()
+#
+# Time Complexity: O(1)
```
Comment (vzvu3k6k) [inline data_structures/arrays/single_number.rb:45]: Not sure this approach is O(1).

I think `Array#find` and `Array#count` iterate over an array and the time complexity of this approach will be O(n^2).

Comment (jsca-kwok) [reply data_structures/arrays/single_number.rb:45]: Ohhh I see! Thank you, I am still new to time complexity and the notation. I will update this.

**Review by vbrazo — APPROVED** [review]

File: data_structures/arrays/single_number.rb (hunk 1 above)
Comment (vbrazo) [reply data_structures/arrays/single_number.rb]: Yeah @jsca-kwok behind the scenes the count and find do a loop.

Comment (vbrazo) [reply data_structures/arrays/single_number.rb]: Count: https://github.com/ruby/ruby/blob/e019dd24df4ed7063ad80d4c2e4070141793f598/array.c#L6084

**Pull request merged**